### ISSUE TITLE
The performance got even worse after training.

### ISSUE THREAD

**Issue by zhangjy328** [Omni6DPose/GenPose2#22]
Dear Author,
I have created my own dataset according to the specified dataset format and successfully completed the training process. However, the performance of the trained model turned out to be **significantly worse**—to an absurdly poor extent. I have no idea what factors might be causing this outcome, and I suspect it could be related to the dataset.

Here is what I have already done:
The unit of the **depth** map is set to **meters (m)**.
The **mask** map correctly distinguishes different object instances.
The remaining issues may lie in the **meta.json** file. All units in my meta.json are also in meters. My scenario involves a single type of object appearing repeatedly in images, with a variable number of objects per image. I assigned different names to each object instance (e.g., `1_object1_object1`), and the mask IDs are aligned with the `id` field in meta.json. However, the `class_name`, `class_label`, `instance_path`, `bbox_side_len`, and `oid` fields are identical for all object instances in the file. Additionally, all the parameters at the bottom of the file—except for the `camera`—were filled in arbitrarily, and most of them were set to `none`. I wonder if this would negatively impact the training process?

I noticed that the `scale `values in the `meta` of your dataset are all around 0.001. What does this `scale `refer to? I set all of mine to 1.0—could this be the key issue? However, I see that the unit of `bbox_side_len `in your dataset should also be meters, so if `bbox_side_len `is in meters, should `scale `still be set to 0.001?

**example：**
```
{
    "objects": {
        "1_object1_object1": {
            "id": 1,
            "meta": {
                "class_name": "object1",
                "class_label": 1,
                "instance_path": ".\\dataset\\ycbv\\models\\object1\\object1.obj",
                "scale": [
                    1.0,
                    1.0,
                    1.0
                ],
                "bbox_side_len": [
                    0.08999999985098839,
                    0.10000000149011612,
                    0.07900000363588333
                ],
                "is_background": false,
                "oid": "object1"
            },
            "quaternion_wxyz": [
                0.6887965798377991,
                -0.17173629999160767,
                0.17039033770561218,
                0.6833981871604919
            ],
            "translation": [
                0.20136845111846924,
                -0.04658902809023857,
                -0.854668915271759
            ],
            "world_quaternion_wxyz": [
                0.9999922513961792,
                0.0,
                0.0,
                0.003934077452868223
            ],
            "world_translation": [
                0.19609126448631287,
                0.36108657717704773,
                0.9235000014305115
            ],
            "is_valid": true,
            "material": [
                "material"
            ]
        },
        "2_object1_object1": {
            "id": 2,
            "meta": {
                "class_name": "object1",
                "class_label": 1,
                "instance_path": ".\\dataset\\ycbv\\models\\object1\\object1.obj",
                "scale": [
                    1.0,
                    1.0,
                    1.0
                ],
                "bbox_side_len": [
                    0.08999999985098839,
                    0.10000000149011612,
                    0.07900000363588333
                ],
                "is_background": false,
                "oid": "object1"
            },
            "quaternion_wxyz": [
                0.6950943470001221,
                -0.17330650985240936,
                0.1687929928302765,
                0.6769915223121643
            ],
            "translation": [
                -0.4121912717819214,
                0.08093658089637756,
                -0.9224755167961121
            ],
            "world_quaternion_wxyz": [
                0.9999129772186279,
                0.0,
                0.0,
                0.013192434795200825
            ],
            "world_translation": [
                0.051659610122442245,
                -0.2524731457233429,
                0.9235000014305115
            ],
            "is_valid": true,
            "material": [
                "material"
            ]
        }
    },
    "camera": {
        "quaternion": [
            0.6861026883125305,
            0.17106463015079498,
            0.17106464505195618,
            0.6861026883125305
        ],
        "translation": [
            0.5561983585357666,
            0.15971815586090088,
            1.7000000476837158
        ],
        "intrinsics": {
            "fx": 387.10170276978414,
            "fy": 247.54346211751303,
            "cx": 387.55371477948086,
            "cy": 316.8295307159424,
            "width": 640,
            "height": 480
        },
        "scene_obj_path": "",
        "background_image_path": "",
        "background_depth_path": "",
        "distances": [
            0.8793057799339294,
            1.0136140584945679
        ],
        "kind": "ycbv_synthetic"
    },
    "env_param": {
        "env_map_id": 4,
        "rotation_euler_z": null,
        "image_value": null,
        "image_path": null
    },
    "face_up": true,
    "concentrated": false,
    "comments": "None",
    "runtime_seed": 59619781,
    "baseline_dis": 0.055,
    "emitter_dist_l": 3,
    "scene_dataset": "GEN"
}

```

**Comment by liukeyuzhenshuai-sys** [comment]
解决了嘛

**Comment by zhangjy328** [comment]
这里大部分参数应该都没用到，长度我用的m为单位

**Comment by liukeyuzhenshuai-sys** [comment]
效果怎么样，推理的资源量是什么样的

**Comment by zhangjy328** [comment]
我找到问题了，是因为我把ema模块注释了，这个模块注释后模型就异常，我暂时也没看明白这个模块起什么作用，但是似乎是不能注释

**Comment by liukeyuzhenshuai-sys** [comment]
<img width="617" height="342" alt="Image" src="https://github.com/user-attachments/assets/e7d8ab5f-c126-4a11-9095-c91811ff0ab5" /> 这是官方的数据集，从2d可视化的角度就很不好，3d可视化的效果更是偏的没谱，不知道你有没有遇到相似的问题

<img width="950" height="868" alt="Image" src="https://github.com/user-attachments/assets/b7989d3c-bcda-4b6c-ac93-5db2c8f7499a" />

**Comment by zhangjy328** [comment]
没训练过公开数据集，太大了，在自己数据集上跑的，位置偏差大概几厘米，角度一两度的样子，self.pts_encoder模块我发现好像毫无用处，我训练的时候`--dino`参数是`none`，推理时直接把`--dino`参数从`none`改成`pointwise`，忽略不匹配层，在完全随机的情况下，推理出来的结果和`none`下推理结果几乎一致，我不理解这段编码增加了384维rgb特征，为什么不用训练就能跑结果

**Comment by zhangjy328** [comment]
我发现，把scorenet里的pts_feat = data['pts_feat']改为全0，pts_feat = torch.zeros(bs, 1024)，你敢相信，丝毫不影响推理结果，这里提取这些特征完全没用啊
